generators/project-name/config.js

```diff
@@ -29,4 +35,4 @@ const defaultConfig = {
   ...requiredConfig,
 };
 
-module.exports = { requiredConfig, defaultConfig };
+module.exports = { requiredConfig, defaultConfig, reproducibleConfig };
```

Comment:
idempotentConfig? defaultConfig? subGeneratorConfig?

Reply:
- requiredConfig needs to be written to .yo-rc.json.
- defaultConfig is not written to .yo-rc.json to keep it clean.
- reproducibleConfig is for testing purpose.

Reply:
Should requiredConfig be merged into defaultConfig?
What names?

Reply:
Guys, instead of naming them this way, we should be naming them by their actual function, not just their properties: how should each variable be used? For example, is it for tests? "testConfig" can be used, or "configForTests".

Reply:
Not every test should use it.
reproducibleConfigForTests would be better.

Reply:
That'd be a better name indeed!